In [2]:
import tensorflow as tf
from keras import datasets,models,layers,losses,optimizers

In [11]:
(x_train, y_train), (x_test, y_test) = datasets.fashion_mnist.load_data()
x_train, x_test = x_train / 255.0, x_test / 255.0
cnn = models.Sequential([
    layers.Conv2D(32, (3,3), activation='relu', input_shape=(28,28,1)),
    layers.MaxPool2D((2,2)),
    layers.Conv2D(64, (3,3), activation='relu'),
    layers.MaxPool2D((2,2)),
    layers.Flatten(),
    layers.Dense(64, activation='relu'),
    layers.Dense(10)
])
cnn.compile(optimizer='adam', loss=losses.SparseCategoricalCrossentropy(from_logits=True), metrics=['accuracy'])
cnn.summary()

Model: "sequential_1"
_________________________________________________________________
 Layer (type)                Output Shape              Param #   
 conv2d_2 (Conv2D)           (None, 26, 26, 32)        320       
                                                                 
 max_pooling2d_2 (MaxPoolin  (None, 13, 13, 32)        0         
 g2D)                                                            
                                                                 
 conv2d_3 (Conv2D)           (None, 11, 11, 64)        18496     
                                                                 
 max_pooling2d_3 (MaxPoolin  (None, 5, 5, 64)          0         
 g2D)                                                            
                                                                 
 flatten_1 (Flatten)         (None, 1600)              0         
                                                                 
 dense_2 (Dense)             (None, 64)               

In [12]:
cnn.fit(x_train, y_train, epochs=10, batch_size=64, verbose=2)

Epoch 1/10
938/938 - 4s - loss: 0.5229 - accuracy: 0.8105 - 4s/epoch - 4ms/step
Epoch 2/10
938/938 - 3s - loss: 0.3452 - accuracy: 0.8757 - 3s/epoch - 3ms/step
Epoch 3/10
938/938 - 3s - loss: 0.3011 - accuracy: 0.8907 - 3s/epoch - 3ms/step
Epoch 4/10
938/938 - 3s - loss: 0.2699 - accuracy: 0.9024 - 3s/epoch - 3ms/step
Epoch 5/10
938/938 - 3s - loss: 0.2452 - accuracy: 0.9107 - 3s/epoch - 4ms/step
Epoch 6/10
938/938 - 3s - loss: 0.2256 - accuracy: 0.9170 - 3s/epoch - 3ms/step
Epoch 7/10
938/938 - 3s - loss: 0.2089 - accuracy: 0.9230 - 3s/epoch - 3ms/step
Epoch 8/10
938/938 - 3s - loss: 0.1932 - accuracy: 0.9282 - 3s/epoch - 3ms/step
Epoch 9/10
938/938 - 3s - loss: 0.1766 - accuracy: 0.9332 - 3s/epoch - 3ms/step
Epoch 10/10
938/938 - 4s - loss: 0.1651 - accuracy: 0.9391 - 4s/epoch - 4ms/step


In [13]:
cnn_loss, cnn_accuracy = cnn.evaluate(x_test, y_test, verbose=1)
print(f'CNN Accuracy on test set: {cnn_accuracy}')

313/313 [==============================] - 1s 3ms/step - loss: 0.2613 - accuracy: 0.9097
CNN Accuracy on test set: 0.9096999764442444


MLP models accuracy is around 88% whereas cnn models accuracy is around 90%.